exercise/templatetags/exercise.py

```diff
@@ -53,6 +54,15 @@ def _get_toc(context, student=None):
 @register.inclusion_tag("exercise/_user_results.html", takes_context=True)
 def user_results(context, student=None):
     values = _get_toc(context, student)
+
+    # Add the difficulty information to all submission to enable
+    # textual status text in points badges in submissions list.
+    for module in values['modules']:
+      for child in module['children']:
```

Comment:
I printed out the data while figuring out how to implement this and it seemed to work with the course that I tested it on. It might not work in all cases though.

Reply:
Thus, my comment. E.g. Tampere is using more levels than our courses do.

Reply:
If I was to attempt to generalize this, is there any documentation/reference regarding the structure of the tree? Is it like this: module -> children -> ... -> children?

Reply:
Vague, but yes. Module is only different layer (shouldn't be, but it's for now) and then all childrens can have more of the same type. You can come and chat with @emmahonkamaa who has the best details of the current configuration hierarchy (except nothing about the cache part). Database structure follows that and then cache does something simplifications on top of it.

Reply:
Just pushed a commit that implements a recursive way of adding difficulty information to submissions - I tested it by adding a child to a child of a module and the info displayed seemed to be fine.